### ISSUE TITLE
Unable to connect to Client - please check the public address

### ISSUE THREAD

**Issue by jervine** [d8ahazard/Phlex#117]
I am running Phlex on a Fedora 26 docker container. I have the application installed and running, however when I go to 'Link Google' the plexserver.cookiehigh.us is unable to connect to my public facing address. I have my Phlex server set up to use https, but I'm using my own CA to sign the certificates, so I'm pretty sure that is the problem I am having. I'm guessing that plexserver.cookiehigh.us will not connect to an untrusted certificate - preventing me from continuing with the configuration.

I expect I can simply expose my Phlex instance over http, although I would probably prefer not to do that. I would also expect that I could potentially use letsencrypt to generate a free SSL certificate for my server, although that would mean setting up my server to use my 'external' domain - something I'd rather avoid.

Are there any other options I can explore here?

**Comment by d8ahazard** [comment]
@jervine 
Actually, self-signed certs should be OK.  You didn't mention - are you port-forwarding to the docker container somewhere?  It does have to be publicly accessible, but I didn't see mention of it in your description.  Or, a better question - can you hit your Phlex server from a cell phone if you go to the URL specified in "Public Address"?

**Comment by d8ahazard** [comment]
Also, in a perfect world, I'd rewrite this in python.  I see you know py.  LMK if you'd be interested in maybe helping me port this.

**Comment by jervine** [comment]
Yes, I can reach my public address, which is at https://phlex.ervine.org/Phlex/ from outside my home network (albeit with certificate errors).

So, port forwarding through the router to the docker server and phlex container is working.

I know enough python to be dangerous, I wouldn't say I 'know it', and would need to try and remember it :-) If you do intend to rewrite in python, I'd try and help as much as I can (or as time allows).

**Comment by d8ahazard** [comment]
Got it.  I'll check the server logs, you should edit your post to remove
the URL.  I'll probably contact you over Hangouts later this evening or
tomorrow to discuss.

**Comment by jervine** [comment]
OK - no worries. Hong Kong timezone if that helps with scheduling.

**Comment by clintbxtreme** [comment]
I'm having the same "Unable to connect to Client" issue. I am running a LAMP server on Ubuntu Mate 16.04. My public address is https://my-domain/phlex. I am able to connect to it outside my network. I have a free letsencrypt SSL certificate. Nothing shows up in the logs.

**Comment by bornsinner84** [comment]
Hello I just tried to get it working but can not link my account ... What is my public address supposed to be? Any help thank you

**Comment by katima-g33k** [comment]
@bornsinner84 the public address is your public IP address (type 'whats my IP address' in Google and you should have it) or a DNS linked to your public IP (dynDNS, no-ip, etc). However I have been trying to get it to work with my DNS and it keeps saying that it can't connect

**Comment by bornsinner84** [comment]
Yeah I can't seem to connect myself. Unable to connect to client

**Comment by jervine** [comment]
Seems to be working now. Guessing there was a change made at the backend?

**Comment by clintbxtreme** [comment]
It's still not working for me. Just tried a fresh setup. I can "Link to Google" using  http://my-domain:8080/phlex (a hole punched through my firewall) but I can't through https://my-domain/phlex. Still says "Unable to connect to Client".

**Comment by Martyfree123** [comment]
So, From what i've seen, It doesn't work with a VPN. I've been getting the same error here that everyone has and finally I realized it was because my IP was being re-routed. I disabled my VPN and now everything works. 

So, I think it's really important to make this work for people with VPNs (Seeing as how most home Media Centres are also Torrent Machines.) There's got to be a way we can fix the issue with talking to the Google Home through a re-routed Public IP.
